plugins/modules/monitoring/uptimerobot.py

```diff
@@ -7,139 +7,183 @@
-    jsonresult = json.loads(result)
-    req.close()
-    return jsonresult['stat']
-
-
-def pauseMonitor(module, params):
-
-    params['monitorStatus'] = 0
-    data = urlencode(params)
-    full_uri = API_BASE + API_ACTIONS['editMonitor'] + data
-    req, info = fetch_url(module, full_uri)
-    result = to_text(req.read())
-    jsonresult = json.loads(result)
-    req.close()
-    return jsonresult['stat']
+class UptimeRobot:
+    def __init__(self, params, api_methods):
+        self.headers = {
+            'Content-Type': "application/json"
+        }
+        self.uri = ''
+        self.api_methods = api_methods
+        self.params = self.check_dict(params)
+        self.body = json.dumps(self.params)
+        for state, method in self.api_methods.items():
+            if state == self.params['status']:
+                self.api_method = method
+
+    def get_monitor_id(self, fname):
+        monitors = self.get_monitors()
+        for m_dict in monitors['monitors']:
+            if m_dict['friendly_name'] == fname:
+                mid = m_dict['id']
+                return mid
+
+    def get_monitors(self):
+        self.body = {}
+        self.body['api_key'] = self.params['api_key']
+        self.body = json.dumps(self.body)
+        self.uri = API_BASE + "getMonitors"
+        state = self.api_call()
+        return state
+
+    def api_call(self):
+        req = Request()
+        state = req.post(url=self.uri, data=self.body, headers=self.headers)
+        json_response = json.loads(to_text(state.read(), errors='surrogate_or_strict'))
+        return json_response
+
+    def api_request(self):
+        self.uri = API_BASE + self.api_method
+        state = self.api_call()
+        return state, self.api_method
+
+    def check_dict(self, dictionary):
+        tmp = {}
+        for k, v in dictionary.items():
+            if v != '':
+                tmp[k] = v
+            if k == 'type':
+                tmp[k] = 3 if v == 'ping' else 1
+
+        if dictionary['status'] != 'created':
+            tmp['id'] = self.get_monitor_id(dictionary['friendly_name'])
+        dictionary.clear()
+        return tmp
 
 
 def main():
+    result = dict(
+        changed=False,
+        original_message='',
+        message=''
+    )
+
+    api_methods = dict(
+        started='editMonitor',
+        present='newMonitor',
+        paused='editMonitor',
+        absent='deleteMonitor',
+        created='newMonitor'
+    )
 
     module = AnsibleModule(
         argument_spec=dict(
-            state=dict(required=True, choices=['started', 'paused']),
+            name=dict(required=True),
+            url=dict(required=True),
+            check_type=dict(required=False, choices=['http', 'ping']),
+            state=dict(required=True, choices=['started', 'paused', 'absent', 'created', 'present']),
             apikey=dict(required=True, no_log=True),
-            monitorid=dict(required=True)
+            monitorid=dict(required=False)
```

Comment:
Is this no longer required, at least for `state='started'` or `state='paused'`? I don't see you ensuring this anywhere.